ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/ImmutableAnalysis.java

```diff
@@ -74,4 +75,6 @@
   AliasedDataSource getFrom();
 
   boolean getOrReplace();
+
+  DataSource getDataSource();
```

Comment:
```suggestion
  DataSource getFromDataSource();
```
in the spirit of bike-shedding, I feel like we shouldn't introduce this extra method since it makes the API a little confusing (when using autocomplete, I'll have to try to understand why we have `getFrom` and `getDataSource` and read the code to work out the difference - spoiler, there isn't one). But if we insist in saving the extra `.getDataSource()` call in the code base, let's try to make it clear that they're really the same method.

Reply:
Ah, I didn't know about `getFrom()`. I also just noticed that this method is unused in this PR, so I should have saved it for the next one. Sorry about that.

I just tried it out on the follow-on branch, and it looks like (Law of Demeter violation notwithstanding) I can just use `.getFrom().getDataSource()` instead.

Reply:
Hmm, now that I'm removing it, I notice that I'd written exactly that code in the implementation, so now I don't know _what_ I was thinking...

Reply:
😆